### PR TITLE
ENT-178: Fix django admin for edx-analytics-data-api

### PR DESCRIPTION
**Pull request by saleem-latif**
__Description:__
This PR movies model, admin and related definitions for user role functionality into a separate app so that proper database is used when operations are performed via these modes.

**TODO Before Merge:**
- [x] Upgrade enterprise data version.
- [x] Add Changelog entry for the changes

**TODO After Merge:**
- [x] Create a release tag for a PyPI release
- [x] Add the newly added app to `settings.INSTALLED_APPS` of `edx-analytics-data-api`

**Testing Instructions:**

- Install `edx-enterprise-data` in `edx-analytics-data-api` using this PR's branch
- Run `edx-analytics-data-api` server
- Visit Django admin portal and click on "Enterprise Data Role Assignment" (or visit http://localhost:8000/site/admin/enterprise_data_roles/enterprisedataroleassignment/)
- Make sure your able to create `Enterprise Data Role Assignment` records without any issue.

### CODE REVIEW

File: enterprise_data_roles/models.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+@python_2_unicode_compatible
+class EnterpriseDataRoleAssignment(UserRoleAssignment):
+    """
+    Model to map users to a EnterpriseDataFeatureRole.
+
+     .. no_pii:
+    """
+
+    role_class = EnterpriseDataFeatureRole
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, db_index=True, on_delete=models.CASCADE)
```
Comment (brittneyexline) [inline enterprise_data_roles/models.py]: nit: this is redundant now that the latest version of edx-rbac should be using settings.AUTH_USER_MODEL for the foreign key reference

**Review by brittneyexline — APPROVED** [review]
looks like you've got some quality things to clean up, but from a code perspective i think it's good to go. is there anything else that we'll need to update in order for it to resolve the django admin issue?

**Comment by saleem-latif** [comment]
@brittneyexline No, Apart from adding the new app to `INSTALLED_APPS`, nothing else is required.

**Review by muhammad-ammar — APPROVED** [review]
LGTM 👍

**Pull request merged**